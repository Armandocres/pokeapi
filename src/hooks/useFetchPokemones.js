import { useState, useEffect } from 'react';
import getPokemones from '../helpers/getPokemones';

const useFetchPokemones = ( pokemon ) => {
  const [state, setstate] = useState({
    data: {},
    loading: true
  });

  useEffect(() => {
		getPokemones(pokemon).then((pokemo) => {
        setstate({
          data: pokemo,
          loading: false
        })
    });
	}, [pokemon]);

  return state;
}

export default useFetchPokemones;