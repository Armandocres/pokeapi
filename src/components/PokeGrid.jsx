import React from "react";
import useFetchPokemones from "../hooks/useFetchPokemones";
import PropTypes from "prop-types";

const PokeGrid = ({ pokemon }) => {
	const { data: pokemones, loading } = useFetchPokemones(pokemon);

	return (
		<div className="card-grid">
			{loading && <p>Cragando...</p>}
			<figure className="card__figure">
				<img src={pokemones.image} alt={pokemones.name} />
			</figure>
			<div>
				<p>Pokemon: {pokemones.name}</p>
				<p>Peso: {pokemones.height}</p>
				<p>Especie: {pokemones.especie}</p>
				<p>Id: {pokemones.id}</p>
			</div>
		</div>
	);
};

PokeGrid.propTypes = {
	pokemon: PropTypes.string.isRequired,
};

export default PokeGrid;
