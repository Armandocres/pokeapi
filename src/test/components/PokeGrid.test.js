import React from 'react';
import { shallow } from 'enzyme';
import PokeGrid from '../../components/PokeGrid';
import useFetchPokemones from '../../hooks/useFetchPokemones';
jest.mock('../../hooks/useFetchPokemones');

describe('Pruebas en componente Pokegrid', () => {
  const name = 'pikachu';

  test('Snapshoot', () => {
    useFetchPokemones.mockReturnValue({
      data: {},
      loading: true,
    })

    const wrapper = shallow(<PokeGrid pokemon={name} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe cambiar el snapshot usando el useFetch', () => {
    const pokemon = {
      id: '1',
      height: '56',
      name: 'pikachu',
      especie: 'pikachu',
      image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/144.svg',
    }

    useFetchPokemones.mockReturnValue({
      data: pokemon,
      loading: false,
    });

    const wrapper = shallow(<PokeGrid pokemon={name} />);
    expect(wrapper).toMatchSnapshot();
  });

});