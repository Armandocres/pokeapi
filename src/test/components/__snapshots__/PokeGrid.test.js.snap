// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente Pokegrid Debe cambiar el snapshot usando el useFetch 1`] = `
<div
  className="card-grid"
>
  <figure
    className="card__figure"
  >
    <img
      alt="pikachu"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/144.svg"
    />
  </figure>
  <div>
    <p>
      Pokemon: 
      pikachu
    </p>
    <p>
      Peso: 
      56
    </p>
    <p>
      Especie: 
      pikachu
    </p>
    <p>
      Id: 
      1
    </p>
  </div>
</div>
`;

exports[`Pruebas en componente Pokegrid Snapshoot 1`] = `
<div
  className="card-grid"
>
  <p>
    Cragando...
  </p>
  <figure
    className="card__figure"
  >
    <img />
  </figure>
  <div>
    <p>
      Pokemon: 
    </p>
    <p>
      Peso: 
    </p>
    <p>
      Especie: 
    </p>
    <p>
      Id: 
    </p>
  </div>
</div>
`;
